zkevm-circuits/src/ecc_circuit.rs

```diff
@@ -0,0 +1,1383 @@
+        log_context_cursor!(ctx);
+
+        let (px, px_cells, px_valid, px_is_zero) =
+            self.precheck_fq(ctx, ecc_chip, op.p.0, powers_of_256);
+        let (py, py_cells, py_valid, py_is_zero) =
+            self.precheck_fq(ctx, ecc_chip, op.p.1, powers_of_256);
+        let p_is_on_curve_or_infinity =
+            self.is_on_curveg1_or_infinity(ctx, ecc_chip, &px, px_is_zero, &py, py_is_zero);
+
+        // point at infinity
+        let infinity = EcPoint::construct(
+            ecc_chip
+                .field_chip()
+                .load_constant(ctx, fe_to_biguint(&Fq::zero())),
+            ecc_chip
+                .field_chip()
+                .load_constant(ctx, fe_to_biguint(&Fq::zero())),
+        );
+        // for invalid case, take a random point.
+        let dummy_g1 = ecc_chip.load_random_point::<G1Affine>(ctx);
+
+        let point_p = EcPoint::construct(px, py);
+        let is_valid = ecc_chip.field_chip().range().gate().and_many(
+            ctx,
+            vec![
+                QuantumCell::Existing(px_valid),
+                QuantumCell::Existing(py_valid),
+                QuantumCell::Existing(p_is_on_curve_or_infinity),
+            ],
+        );
+        let point_p = ecc_chip.select(ctx, &point_p, &dummy_g1, &is_valid);
+
+        let scalar_s = self.handle_fr(ctx, fr_chip, op.s);
+
+        let res = op.r.unwrap_or(G1Affine::identity());
+        let point_r = self.handle_g1(ctx, ecc_chip, res, powers_of_256);
+
+        log::trace!("[ECC] EcMul Inputs Assigned:");
+        log_context_cursor!(ctx);
+
+        let point_r_got = ecc_chip.scalar_mult(
+            ctx,
+            &point_p,
+            &scalar_s.scalar.limbs().to_vec(),
+            fr_chip.limb_bits,
+            4,
+        );
+        let point_r_got = ecc_chip.select(ctx, &point_r_got, &infinity, &is_valid);
+        ecc_chip.assert_equal(ctx, &point_r.ec_point, &point_r_got);
+
+        log::trace!("[ECC] EcMul Assignmnet END:");
+        log_context_cursor!(ctx);
+
+        EcMulDecomposed {
+            is_valid,
+            point_p: G1Decomposed {
+                ec_point: point_p,
+                x_cells: px_cells,
+                y_cells: py_cells,
+            },
+            scalar_s,
+            point_r,
+        }
+    }
+
+    /// Decomposes an EcPairing operation and returns cells that represent the LE-bytes of all
+    /// (G1, G2) pairs. In phase2 they will be RLC'd with the keccak randomness.
+    #[allow(clippy::too_many_arguments)]
+    fn decompose_ec_pairing_op(
+        &self,
+        ctx: &mut Context<F>,
+        ecc_chip: &EccChip<F, FpConfig<F, Fq>>,
+        _fr_chip: &FpConfig<F, Fr>,
+        pairing_chip: &PairingChip<F>,
+        fp12_chip: &Fp12Chip<F, FpConfig<F, Fq>, Fq12, XI_0>,
+        powers_of_256: &[QuantumCell<F>],
+        op: &EcPairingOp,
+    ) -> EcPairingDecomposed<F> {
+        log::trace!("[ECC] ==> EcPairing Assignment START:");
```

Comment:
Nitpick: same as above

Reply:
fixed in [4d5070f](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1720/commits/4d5070fdf1d7821cab0b4b2caf5f0582fe291ae2)